### PR TITLE
Support skipping `python_min` and `pip_no_build_backend` hints

### LINKED ISSUE #2170: ENH: Allow ignoring `python_min` hints
Some feedstocks need to opt-out of the test pins for `python_min`, for example due to being used elsewhere as a `downstreams:` test against other python versions (c.f. https://github.com/conda-forge/conda-forge.github.io/pull/2387, https://github.com/conda-forge/conda-smithy-feedstock/pull/331, https://github.com/conda-forge/setuptools-feedstock/pull/371).

Given this reality, we shouldn't force those feedstocks to live with linter warnings in perpetuity, but give them a way to silence the linter. In https://github.com/conda-forge/conda-smithy/pull/2090, the following mechanism was introduced
```yaml
linter:
  skip:
    - lint_noarch_selectors
```

This could easily be extended to allow
```yaml
linter:
  skip:
    - hint_python_min
```

### PR DESCRIPTION
**Pull request by mgorny**
Checklist
* [x] Added a ``news`` entry
* [x] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

Fixes #2170 

Add options to skip linter checks for `python_min` and missing build backend, both requested in #2170. While at it, fix a `TypeError` that occurs when `conda-forge.yml` contains empty `linter:` or `skip:` keys.

### CODE REVIEW

File: tests/test_lint_recipe.py [hunk 1]
```diff
@@ -3313,23 +3333,38 @@ def test_hint_pip_no_build_backend(
...
+@pytest.mark.parametrize("skip", [False, True])
 def test_hint_noarch_python_use_python_min(
     meta_str,
     expected_hints,
+    skip,
+    tmp_path,
 ):
+    if skip:
+        if not expected_hints:
+            pytest.skip("No hints expected")
```
Comment (h-vetinari) [inline tests/test_lint_recipe.py:3345]: Why should we need to skip this test if there are no expected hints? We could reformulate the test below to expect empty hints, no?

File: tests/test_lint_recipe.py [hunk 2]
```diff
@@ -3313,23 +3333,38 @@ def test_hint_pip_no_build_backend(
...
         hints,
         recipe_version=0,
     )
 
     # make sure we have the expected hints
-    if expected_hints:
+    if expected_hints and not skip:
         for expected_hint in expected_hints:
             assert any(expected_hint in hint for hint in hints), hints
     else:
```
Comment (h-vetinari) [inline tests/test_lint_recipe.py:3370]: AFAIU the `else:` branch here doesn't actually check that the hint-to-be-skipped is not present, it just checks that a specific hint is not contained.

**Review by h-vetinari — COMMENTED** [review]
Thanks for the PR!

I have just one point: I think `test_hint_noarch_python_use_python_min` should become more similar in logic to `test_hint_pip_no_build_backend`, as the testing in the former is currently not as good as in the latter.

File: tests/test_lint_recipe.py (hunk 1 above)
Comment (mgorny) [reply tests/test_lint_recipe.py:3345]: It always expected empty hints. I've skipped it because it's redundant, i.e. doesn't test anything.

File: tests/test_lint_recipe.py (hunk 2 above)
Comment (mgorny) [reply tests/test_lint_recipe.py:3370]: I'm sorry but I don't understand this sentence. The `else:` branch checks whether any of the hints generated matches `python_min` hints. FWIU this is exactly what we want to test for here.

Comment (h-vetinari) [reply tests/test_lint_recipe.py:3370]: yeah, I got confused there. Still, the test could be sharpened to my mind, in the sense that we could _exactly_ expect the set of hints to be `expected_hints - noarch_hint * skip`

File: tests/test_lint_recipe.py (hunk 1 above)
Comment (mgorny) [reply tests/test_lint_recipe.py:3345]: That makes sense, I'll update it. In the end, the skips don't save that much time, and could be confusing.

**Review by h-vetinari — APPROVED** [review]

**Comment by mgorny** [comment]
Thanks!
